src/mono/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs

```diff
@@ -204,8 +196,7 @@ public void Interrupt()
 
         public bool Join(int millisecondsTimeout)
         {
-            if (millisecondsTimeout < Timeout.Infinite)
-                throw new ArgumentOutOfRangeException(nameof(millisecondsTimeout), millisecondsTimeout, SR.ArgumentOutOfRange_NeedNonNegOrNegative1);
+            VerifyTimeoutMilliseconds(millisecondsTimeout);
             return JoinInternal(this, millisecondsTimeout);
```

Comment:
This can be `return JoinInternal(this, VerifyTimeoutMilliseconds(millisecondsTimeout));` if you keep the VerifyTimeoutMilliseconds method.